packages/uicore/src/components/TableV3/TableV3.tsx

```diff
@@ -0,0 +1,218 @@
+      ? '-4px 0 4px -4px gray inset'
+      : isFirstRightPinnedColumn
+      ? '4px 0 4px -4px gray inset'
+      : undefined,
+    left: isPinned === 'left' ? `${column.getStart('left')}px` : undefined,
+    right: isPinned === 'right' ? `${column.getAfter('right')}px` : undefined,
+    position: isPinned ? 'sticky' : 'relative',
+    width: column.getSize(),
+    zIndex: isPinned ? 1 : 0
+  }
+}
+
+interface ExecutionListV2Props<Data extends Record<string, unknown>> {
+  leftFixed?: string[]
+  rightFixed?: string[]
+  tableData: Data[]
+  columns: Array<ColumnDef<unknown>>
+  resize?: boolean
+  className?: string
+  hideHeaders?: boolean
+  initialState?: Partial<
+    VisibilityTableState &
+      ColumnOrderTableState &
+      ColumnPinningTableState &
+      SortingTableState &
+      ExpandedTableState &
+      GroupingTableState &
+      ColumnSizingTableState &
+      PaginationTableState &
+      RowSelectionTableState
+  >
+  name?: string
+}
+
+const DEFAULT_EMPTY_ARR: string[] = []
+
+export const TableV3 = <Data extends Record<string, unknown>>(
+  props: ExecutionListV2Props<Data>
+): React.ReactElement => {
+  const containerRef = useRef(null)
+  const {
+    tableData = DEFAULT_EMPTY_ARR,
+    columns = [],
+    resize = false,
+    hideHeaders = false,
+    initialState = {},
+    name
+  } = props
+
+  const [sorting, setSorting] = React.useState<SortingState>([])
+
+  const table = useReactTable({
+    data: defaultTo(tableData, []),
+    columns: columns,
+    columnResizeMode: 'onChange',
+    getCoreRowModel: getCoreRowModel(),
+
+    getSortedRowModel: getSortedRowModel(),
+    state: {
+      sorting
+    },
+    onSortingChange: setSorting,
+    initialState: {
+      ...initialState
+    }
+  })
+
+  return (
+    <div
+      id="execution-table-container"
+      ref={containerRef}
+      style={{
+        direction: table?.options?.columnResizeDirection
+      }}>
+      <div
+        style={{
+          maxWidth: defaultTo(get(containerRef, 'current.offsetWidth'), '100%')
+        }}
+        className={css.tableContainer}>
+        <table
+          {...{
+            style: {
+              overflowX: 'scroll',
+              width: table.getCenterTotalSize()
+            }
+          }}>
+          {hideHeaders ? null : (
+            <thead>
+              {table.getHeaderGroups().map(headerGroup => (
+                <tr key={headerGroup?.id}>
+                  {headerGroup.headers.map(header => {
+                    const { column } = header
+                    const isResizing = header.column.getIsResizing()
+                    const resizeOptions = resize
+                      ? {
+                          onDoubleClick: () => header.column.resetSize(),
+                          onMouseDown: header.getResizeHandler(),
+                          onTouchStart: header.getResizeHandler(),
+                          className: `${css.resizer} ${table.options.columnResizeDirection} ${cx({
```

Comment:
Should be: className={cx(css.resizer, table.options.columnResizeDirection, {[css.isResizing]: isResizing})}

Reply:
improved